Fix building with -fPIC
http://bugs.gentoo.org/show_bug.cgi?id=34712
(note: this is outdated since we fixed the PIC issues in our linux-headers 
ebuilds when doing syscall()'s on x86)

Fix running on amd64
http://bugs.gentoo.org/show_bug.cgi?id=93694

--- raidtools-1.00.3/raid_io.c
+++ raidtools-1.00.3/raid_io.c
@@ -81,7 +81,7 @@
 }
 #undef F
 
-#if !(defined(__alpha__) || defined(__sparc_v9__) || defined(__ia64__))
+#if !(defined(__alpha__) || defined(__sparc_v9__) || defined(__ia64__) || defined(__x86_64__) || defined(__PIC__))
 # ifndef __NR__llseek
 #  ifdef __sparc__
 #   define __NR__llseek		236
@@ -105,7 +105,7 @@
 	long long result;
 	int retval;
 
-#if defined(__alpha__) || defined(__sparc_v9__) || defined(__ia64__)
+#if defined(__alpha__) || defined(__sparc_v9__) || defined(__ia64__) || defined(__x86_64__) || defined(__PIC__)
 	return lseek(fd, offset, SEEK_SET);
 #else
 	retval = _llseek (fd, ((unsigned long long) offset) >> 32,
